mod-pubsub-server/src/main/java/org/folio/services/impl/PublishingServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+package org.folio.services.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.kafka.client.producer.KafkaProducer;
+import io.vertx.kafka.client.producer.impl.KafkaProducerRecordImpl;
+import org.folio.kafka.PubSubConfig;
+import org.folio.rest.jaxrs.model.AuditMessage;
+import org.folio.rest.jaxrs.model.AuditMessagePayload;
+import org.folio.rest.jaxrs.model.Event;
+import org.folio.rest.util.MessagingModuleFilter;
+import org.folio.services.AuditMessageService;
+import org.folio.services.MessagingModuleService;
+import org.folio.services.PublishingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.BadRequestException;
+
+import java.util.Date;
+import java.util.UUID;
+
+import static java.lang.String.format;
+import static org.folio.rest.jaxrs.model.MessagingModule.ModuleRole.PUBLISHER;
+import static org.folio.rest.jaxrs.model.MessagingModule.ModuleRole.SUBSCRIBER;
+
+@Component
+public class PublishingServiceImpl implements PublishingService {
```

Comment:
The implementation is tightly coupled to the KafkaProducer, let's rename it from 'PublishingServiceImpl' to 'KafkaPublisherService'

Reply:
I agree. We can use "interface PublishingService" with several implementations. And KafkaPublisherServiceImpl seems like a good naming for it

Reply:
done, thank you